### PR TITLE
New public dashboard fav map

### LINKED ISSUE #2043: Fav map header for new public dashboard
Milestone for #2030 

screen shot 2015-02-03 at 14 15 55

- [x] Render viewed user's visualization that has most views
- [x] Render spinner until map is loaded
- [x] If no visualizations render a default fallback map (like done in private dashboard)

### PR DESCRIPTION
**Pull request by viddo**
Fixed #2043

- Since both onboarding and the public pages should have the same default fallback map I've refactored the onboarding view slightly.
- The public pages now will be rendered like in screenshots:

screen shot 2015-02-03 at 18 51 29
screen shot 2015-02-03 at 18 51 42

### CODE REVIEW

**Comment by viddo** [comment]
ping @xavijam @saleiva and yes, I know that the maps should be rendered to the left with placeholders (will tackle that later) so need to mention that.. ;)

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1409/))

**Comment by saleiva** [comment]
Looks great!!!! :)

File: lib/assets/javascripts/cartodb/new_common/views/create_default_fallback_map.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+var L = require('leaflet');
```
Comment (javisantana) [inline lib/assets/javascripts/cartodb/new_common/views/create_default_fallback_map.js]: no leaflet code should be placed in cartodb, this should be done using cartodb.js abstractions or using a fake viz.json

Comment (viddo) [inline lib/assets/javascripts/cartodb/new_common/views/create_default_fallback_map.js]: I refactored the existing code related to this, I simply defined the [defined the dependency as a browserify-shim named 'leaflet' in package.json](https://github.com/CartoDB/cartodb/blob/2043-new-public-dashboard-fav-map/package.json#L76) to make the dependency clear. AFAIK, no actual leaflet code has been added (can you confirm @xavijam ?). What would be a better shim name to make it clearer what `window.L` object is?

Comment (javisantana) [inline lib/assets/javascripts/cartodb/new_common/views/create_default_fallback_map.js]: L.map is leaflet code, no references to leaflet should be placed in cartodb, It's not a matter of browserify or dependencies, it's a matter of abstract cartodb from the mapping library you are using

Comment (viddo) [inline lib/assets/javascripts/cartodb/new_common/views/create_default_fallback_map.js]: Ok, thanks for the clarification, I'll take care of it

Comment (javisantana) [inline lib/assets/javascripts/cartodb/new_common/views/create_default_fallback_map.js]: here you have a simple viz.json

https://gist.github.com/javisantana/5423e528e8761b73fcce

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1412/))

**Comment by viddo** [comment]
Using CartoDB.js to render the fallback map as well, please let me know if you have more doubts or if it's OK to merge

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1415/))

File: lib/assets/javascripts/cartodb/new_dashboard/onboarding_view.js [hunk 2]
```diff
@@ -49,7 +52,9 @@ module.exports = cdb.core.View.extend({
   },
 
   _hideWelcome: function(e) {
-    if (e) e.preventDefault();
+    if (e) {
+      e.preventDefault();
+    }
     this.localStorage.set({ 'dashboard.welcome': true });
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/new_dashboard/onboarding_view.js]: Why? :DDD Don't you like if in the same line?

**Comment by xavijam** [comment]
:+1:

File: lib/assets/javascripts/cartodb/new_dashboard/onboarding_view.js (hunk 2 above)
Comment (viddo) [inline lib/assets/javascripts/cartodb/new_dashboard/onboarding_view.js]: I agree with sentiments written down at https://github.com/Vizzuality/cartodb-management/wiki/CartoDB-JavaScript-Style-Guide#blocks ;)

Comment (xavijam) [inline lib/assets/javascripts/cartodb/new_dashboard/onboarding_view.js]: jur, ok, another thing to change, go ahead!

**Pull request merged**